Testall fails in Version 0.5.0-dev+2533

While running make testall in:
```
julia> versioninfo()
Julia Version 0.5.0-dev+2533
Commit 69f7571* (2016-02-08 05:36 UTC)
Platform Info:
  System: Linux (x86_64-suse-linux)
  CPU: Intel(R) Core(TM) i5-4460  CPU @ 3.20GHz
  WORD_SIZE: 64
  BLAS: libopenblas (USE64BITINT DYNAMIC_ARCH NO_AFFINITY Haswell)
  LAPACK: libopenblas64_
  LIBM: libopenlibm
  LLVM: libLLVM-3.7.1
```
Testall fails with:

```
	From worker 5:	     * replcompletions       in   6.34 seconds, maxrss  839.86 MB
	From worker 5:	     * replutil             Test Failed
	From worker 5:	  Expression: contains(err_str,"Exponentiation yielding a complex result requires a complex argument")
	From worker 2:	     * repl                  in   9.81 seconds, maxrss  800.62 MB


From worker 6:	     * examples              in  12.31 seconds, maxrss  304.92 MB
	From worker 3:	     * subarray              in 434.34 seconds, maxrss  978.30 MB
	From worker 4:	     * misc                  in 129.50 seconds, maxrss 1023.04 MB
Exception running test replutil :
On worker 5:
LoadError: There was an error during testing
 in record at ./test.jl:317
 in do_test at ./test.jl:215
 [inlined code] from ./essentials.jl:80
 in include_string at ./loading.jl:371
 in include_from_node1 at ./loading.jl:420
 [inlined code] from ./util.jl:179
 in runtests at /home/colin/Downloads/julia/test/testdefs.jl:7
 in #16 at /home/colin/Downloads/julia/test/runtests.jl:36
 in run_work_thunk at ./multi.jl:714
 [inlined code] from ./multi.jl:1010
 in #253 at ./task.jl:59
while loading /home/colin/Downloads/julia/test/replutil.jl, in expression starting on line 170
ERROR: LoadError: Some tests exited with errors.
 in (::##11#19)() at /home/colin/Downloads/julia/test/runtests.jl:64
 in cd(::##11#19, ::ASCIIString) at ./file.jl:47
 in include(::ASCIIString) at ./boot.jl:264
 in include_from_node1(::UTF8String) at ./loading.jl:417
 in process_options(::Base.JLOptions) at ./client.jl:244
 in _start() at ./client.jl:300
while loading /home/colin/Downloads/julia/test/runtests.jl, in expression starting on line 13
Makefile:12: recipe for target 'all' failed
make[1]: *** [all] Error 1
Makefile:572: recipe for target 'testall' failed
make: *** [testall] Error 2

```

--- a/base/replutil.jl
+++ b/base/replutil.jl
@@ -126,7 +126,7 @@ function showerror(io::IO, ex::DomainError, bt; backtrace=true)
                 print(io,"\n$(code.func) will only return a complex result if called with a complex argument. Try $(string(code.func))(complex(x)).")
             elseif (code.func == :^ && code.file == symbol("intfuncs.jl")) || code.func == :power_by_squaring #3024
                 print(io, "\nCannot raise an integer x to a negative power -n. \nMake x a float by adding a zero decimal (e.g. 2.0^-n instead of 2^-n), or write 1/x^n, float(x)^-n, or (x//1)^-n.")
-            elseif code.func == :^ && (code.file == symbol("promotion.jl") || code.file == symbol("math.jl"))
+            elseif code.func == :^ && (code.file == symbol(joinpath(".","promotion.jl")) || code.file == symbol(joinpath(".","math.jl")))
                 print(io, "\nExponentiation yielding a complex result requires a complex argument.\nReplace x^y with (x+0im)^y, Complex(x)^y, or similar.")
             end
             break

--- a/test/replutil.jl
+++ b/test/replutil.jl
@@ -170,16 +170,16 @@ let undefvar
     @test contains(err_str, "Exponentiation yielding a complex result requires a complex argument")
 
     err_str = @except_str [5,4,3][-2,1] BoundsError
-    @test err_str == "BoundsError: attempt to access 3-element Array{Int64,1}:\n 5\n 4\n 3\n  at index [-2,1]"
+    @test err_str == "BoundsError: attempt to access 3-element Array{$Int,1}:\n 5\n 4\n 3\n  at index [-2,1]"
     err_str = @except_str [5,4,3][1:5] BoundsError
-    @test err_str == "BoundsError: attempt to access 3-element Array{Int64,1}:\n 5\n 4\n 3\n  at index [1:5]"
+    @test err_str == "BoundsError: attempt to access 3-element Array{$Int,1}:\n 5\n 4\n 3\n  at index [1:5]"
 
     err_str = @except_str 0::Bool TypeError
-    @test err_str == "TypeError: non-boolean (Int64) used in boolean context"
+    @test err_str == "TypeError: non-boolean ($Int) used in boolean context"
     err_str = @except_str 0::AbstractFloat TypeError
-    @test err_str == "TypeError: typeassert: expected AbstractFloat, got Int64"
+    @test err_str == "TypeError: typeassert: expected AbstractFloat, got $Int"
     err_str = @except_str 0::7 TypeError
-    @test err_str == "TypeError: typeassert: expected Type{T}, got Int64"
+    @test err_str == "TypeError: typeassert: expected Type{T}, got $Int"
     err_str = @except_str "" <: AbstractString TypeError
     @test err_str == "TypeError: subtype: expected Type{T}, got ASCIIString"
     err_str = @except_str AbstractString <: "" TypeError
@@ -242,7 +242,7 @@ let err_str,
     err_str = @except_str FunctionLike()() MethodError
     @test contains(err_str, "MethodError: no method matching (::FunctionLike)()")
     err_str = @except_str [1,2](1) MethodError
-    @test contains(err_str, "MethodError: objects of type Array{Int64,1} are not callable\nUse square brackets [] for indexing an Array.")
+    @test contains(err_str, "MethodError: objects of type Array{$Int,1} are not callable\nUse square brackets [] for indexing an Array.")
     # Issue 14940
     err_str = @except_str randn(1)() MethodError
     @test contains(err_str, "MethodError: objects of type Array{Float64,1} are not callable")

